### PR TITLE
add 'linesglues' debug messages

### LINKED ISSUE #994: integrate line stretching in debug messages
This patch:

```diff
diff --git a/tex/gregoriotex.lua b/tex/gregoriotex.lua
index e8b21b7..fe1687f 100644
--- a/tex/gregoriotex.lua
+++ b/tex/gregoriotex.lua
@@ -356,6 +356,8 @@ local function process (h, groupcode, glyphes)
         --log("eating line")
         h, line = remove(h, line)
       else
+        local glue_sign_name = {[0] = 'normal', [1] = 'stretching', [2] = 'shrinking'}
+        texio.write_nl(string.format('new line: %s factor %f', glue_sign_name[line.glue_sign], line.glue_set))
         centerstartnode = nil
         line_id = nil
         line_top = nil
```

prints debug messages about the stretching/shrinking factors of the different lines. This would be very useful to trigger in the debug interface.

### PR DESCRIPTION
**Pull request by eroux**
Fixes #994

### CODE REVIEW

**Comment by rpspringuel** [comment]
Two things:

1. Under what conditions should I expect to find this debug message being triggered?  I haven't been able to trigger it with a couple of quick tests that I had readily available.
2. It doesn't look like you've handled the situation where the `\gre@debug` is `all`.  When that is the case all debug messages should print, regardless of type.

**Comment by eroux** [comment]
Thanks, these are addressed, I think you didn't see it because the format was inconsistent.

File: tex/gregoriotex.lua [hunk 1]
```diff
@@ -327,6 +329,23 @@ local function center_translation(startnode, endnode, ratio, sign, order)
   startnode.head.next.next.kern = -X
 end
 
+local debug_types_activated = {['linesglues'] = false}
+
+local function set_debug_string(debugstring)
+  texio.write_nl(debugstring)
```
Comment (rpspringuel) [inline tex/gregoriotex.lua]: This line is creating an extra "linesglue" in the log just before the "GregorioTeX is in debug mode..." message.  Do you really need it given that said message already indicates which messages are being printed?

**Comment by rpspringuel** [comment]
I still must be using a bad example for this because I'm not triggering the message.  Can you point to a test in the repository which you expect to trigger this message?

**Comment by eroux** [comment]
Compiling the file

```tex
\documentclass[12pt, a5paper]{book}
\usepackage[autocompile,debug=linesglues]{gregoriotex}
\begin{document}
\gabcsnippet{(c4) Pro(d)phé(ddo>)tæ(c) pre(d)di(f)ca(e)vé(d)runt(dc~) *(,) na(f)sci(g) Dó(h)mi(g)num(i) nos(h)ter(gf) Sal(h>)va(g)tó(f_0fg)rem(f_) (;) de(f) Vír(f)gi(g)ne(e) Ma(f)rí(d.)a.(d.) (::) <eu>E(h) u(h) o(g) u(f) a(gh) e</eu>(g) (::)}
\end{document}
```

I'm getting

```
...
GregorioTeX is in debug mode
linesglues messages will be printed to the log.


LaTeX Font Warning: Font shape `OT1/cmr/m/n' in size <40> not available
(Font)              size <24.88> substituted on input line 12.

GregorioTeX debug: (linesglues): line 1: stretching factor 22%
GregorioTeX debug: (linesglues): line 2: stretching factor 61%
GregorioTeX debug: (linesglues): line 3: stretching factor 17738% [1{/var/lib/te
xmf/fonts/map/pdftex/updmap/pdftex.map}] (./984.aux)

LaTeX Font Warning: Size substitutions with differences
...
```

**Comment by rpspringuel** [comment]
I'm not getting that.  Let my try reinstalling and I'll get back to you.

**Comment by rpspringuel** [comment]
That was the problem: had the wrong branch installed.  This looks good.

**Pull request merged**